### DISCUSSION TITLE
GeometricElement --> InformationElement

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3303] [Q&A] [answered]
hello good evening everyone,
From what I was able to understand about the structure
"bis" -> "Information element" contains information shared by various elements
in my facility I have
1) IFCDynamic.IfcColumn
2) IFCDynamic.IfcColumnType
are the IfcColumnType linked with the GeometricElement?

when I do a search with ecsql for a particular field of the IfcColumnType I get only one element that I can't connect with the geometric element ..

![ecsql1 pgn](https://user-images.githubusercontent.com/40106886/156935429-5ad45787-8a63-4c53-8aac-7bbc9c570c32.png)

![ecsql2](https://user-images.githubusercontent.com/40106886/156935434-863e95a3-bd27-4bf9-893b-4c18e29671e8.png)

Thanks for the help and sorry for everything

**Comment by grigasp** [comment, marked as answer]
If I'm not mistaken, `IFCDynamic.IfcColumnType` derives from the [bis.TypeDefinitionElement](https://github.com/iTwin/bis-schemas/blob/91c5198ac8ca53dff282775e20b756e6fa01fd97/Domains/Core/Released/BisCore.01.00.14.ecschema.xml#L604-L608) class. Type definition elements are linked to geometric elements through either a subclass of [bis.GeometricElement2dHasTypeDefinition](https://github.com/iTwin/bis-schemas/blob/91c5198ac8ca53dff282775e20b756e6fa01fd97/Domains/Core/Released/BisCore.01.00.14.ecschema.xml#L667-L675) or a subclass of [bis.GeometricElement3dHasTypeDefinition](https://github.com/iTwin/bis-schemas/blob/91c5198ac8ca53dff282775e20b756e6fa01fd97/Domains/Core/Released/BisCore.01.00.14.ecschema.xml#L686-L694) depending on whether the element is 2d or 3d. You can also find some more details about type definitions in the [BIS Type Definitions page](https://www.itwinjs.org/bis/intro/type-definitions/).
